### PR TITLE
End of meeting indication

### PR DESCRIPTION
**Pull request by jtkpiotr**
@lkudyba @quirk0o @mhib @adamzima @pbzymek @s1mplex

### CODE REVIEW

File: client/app/components/conference_room/event/CurrentEvent.jsx [hunk 1]
```diff
@@ -31,49 +33,95 @@ Event.propTypes = {
...
+
+  _timeProgressComponent() {
+    if (this.props.event || this.props.nextEventStart) {
+      return (
+        <TimeProgressContainer event={this.props.event}
+                               nextEventStart={this.props.nextEventStart}
+                               displayEndTimeWarning={this._displayEndTimeWarning}
+                               onCompleted={this._onCompleted} />);
+    }
+    return '';
```
Comment (madejejej) [inline client/app/components/conference_room/event/CurrentEvent.jsx:69]: It's probably better to return `null` here

File: client/app/components/conference_room/event/CurrentEvent.jsx [hunk 2]
```diff
@@ -31,49 +33,95 @@ Event.propTypes = {
 
 const NoEvent = () => <p>No event currently in progress</p>;
 
-const TimeProgressContainer = ({ event, nextEventStart, onCompleted }) => (
+const TimeProgressContainer = ({ event, nextEventStart, onCompleted, displayEndTimeWarning }) => (
```
Comment (madejejej) [inline client/app/components/conference_room/event/CurrentEvent.jsx]: Why do you keep multiple components in one file?

File: client/app/components/shared/time/TimeProgress.jsx [hunk 3]
```diff
@@ -37,14 +41,24 @@ export default class TimeProgress extends React.Component {
...
   }
 
   _update() {
-    this.setState({ timeLeft: this._timeLeft() });
-    if (this.props.onCompleted && this.props.end <= moment()) {
+    const timeToEnd = this._timeToEnd();
+
+    this.setState({ timeLeft: this._humanizedTimeLeft() });
+    if (timeToEnd <= 0) {
       this.props.onCompleted();
```
Comment (madejejej) [inline client/app/components/shared/time/TimeProgress.jsx:57]: Does it make sense to clear the interval here?

File: client/app/components/conference_room/event/CurrentEvent.jsx [hunk 4]
```diff
@@ -31,49 +33,95 @@ Event.propTypes = {
...
-};
+  static defaultProps = {
+    onCompleted: () => {}
+  };
+
+  constructor(...args) {
+    super(...args);
+
+    bindAll(this, ['_displayEndTimeWarning', '_onCompleted']);
+    this.state = { endTimeWarning: false };
```
Comment (madejejej) [inline client/app/components/conference_room/event/CurrentEvent.jsx]: What do you think about renaming this to `displayingEndTimeWarning`?

File: client/app/components/conference_room/event/CurrentEvent.jsx (hunk 2 above)
Comment (jtkpiotr) [reply client/app/components/conference_room/event/CurrentEvent.jsx]: done

File: client/app/components/conference_room/event/CurrentEvent.jsx (hunk 4 above)
Comment (jtkpiotr) [reply client/app/components/conference_room/event/CurrentEvent.jsx]: done

File: client/app/components/shared/time/TimeProgress.jsx (hunk 3 above)
Comment (jtkpiotr) [reply client/app/components/shared/time/TimeProgress.jsx:57]: Done

**Review by madejejej — APPROVED** [review]

**Pull request merged**